# Construction of pYPK0_PGI1tp_ScXKS1_FBA1tp (pMEC1129)

[pYPKa_Z_PGI1tp](pYPKa_Z_PGI1tp.ipynb)

[pYPKa_E_FBA1tp](pYPKa_E_FBA1tp.ipynb)

In [1]:
import pydna
from pydna import ipynb_importer

In [2]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [3]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [4]:
pYPK0 = pydna.read("pYPK0.gb")

In [5]:
promoter_clone = pYPKa_Z_PGI1tp = pydna.read("pYPKa_Z_PGI1tp.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nucleotide/NM_001181323.3) record

In [7]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/9169869 width=100% height=250></iframe>')

In [8]:
gb = pydna.Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("NM_001181323.3")

In [9]:
gene_template.list_features()

+----------+-----------+-------+------+--------+--------------+--------+------+
| Feature# | Direction | Start | End  | Length | id           | type   | orf? |
+----------+-----------+-------+------+--------+--------------+--------+------+
| 0        |    -->    |   0   | 1803 |   1803 | <unknown id> | source | yes  |
| 1        |    -->    |   0   | 1803 |   1803 | <unknown id> | gene   | yes  |
| 2        |    -->    |   0   | 1803 |   1803 | <unknown id> | CDS    | yes  |
+----------+-----------+-------+------+--------+--------------+--------+------+

In [10]:
orf = gene_template.extract_feature(2)

In [11]:
orf.isorf()

True

In [12]:
print len(orf)
orf.seguid()

1803


27N2cQNKNSUXfhxo37o-d6AshX0

In [13]:
print str(orf.seq.translate())

MLCSVIQRQTREVSNTMSLDSYYLGFDLSTQQLKCLAINQDLKIVHSETVEFEKDLPHYHTKKGVYIHGDTIECPVAMWLEALDLVLSKYREAKFPLNKVMAVSGSCQQHGSVYWSSQAESLLEQLNKKPEKDLLHYVSSVAFARQTAPNWQDHSTAKQCQEFEECIGGPEKMAQLTGSRAHFRFTGPQILKIAQLEPEAYEKTKTISLVSNFLTSILVGHLVELEEADACGMNLYDIRERKFSDELLHLIDSSSKDKTIRQKLMRAPMKNLIAGTICKYFIEKYGFNTNCKVSPMTGDNLATICSLPLRKNDVLVSLGTSTTVLLVTDKYHPSPNYHLFIHPTLPNHYMGMICYCNGSLARERIRDELNKERENNYEKTNDWTLFNQAVLDDSESSENELGVYFPLGEIVPSVKAINKRVIFNPKTGMIEREVAKFKDKRHDAKNIVESQALSCRVRISPLLSDSNASSQQRLNEDTIVKFDYDESPLRDYLNKRPERTFFVGGASKNDAIVKKFAQVIGATKGNFRLETPNSCALGGCYKAMWSLLYDSNKIAVPFDKFLNDNFPWHVMESISDVDNENWDRYNSKIVPLSELEKTLI*


In [14]:
pf, pr = pydna.parse('''>XX_YPK_rec_fwd 
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTTGTGTTCAGTAATTCAGAG
                        >XK_YPK_rec_rev 
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgTTAGATGAGAGTCTTTTCCAGTTC''')

In [15]:
g = pydna.pcr( pf, pr, orf)

In [16]:
g.figure()

                                        5ATGTTGTGTTCAGTAATTCAGAG...GAACTGGAAAAGACTCTCATCTAA3
                                                                   |||||||||||||||||||||||| tm 50.3 (dbd) 60.9
                                                                  3CTTGACCTTTTCTGAGAGTAGATTgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTTGTGTTCAGTAATTCAGAG3
                                         ||||||||||||||||||||||| tm 49.1 (dbd) 59.0
                                        3TACAACACAAGTCATTAAGTCTC...CTTGACCTTTTCTGAGAGTAGATT5

In [17]:
terminator_clone = pYPKa_E_FBA1tp = pydna.read("pYPKa_E_FBA1tp.gb")

In [19]:
p  = pydna.pcr( p167, p567, promoter_clone)
t  = pydna.pcr( p568, p166, terminator_clone)

In [20]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [21]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Amplicon(1161), Amplicon(1881), Amplicon(736))

In [22]:
asm = pydna.Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [23]:
asm

Assembly:
Sequences........................: [5681] [1161] [1881] [736]
Sequences with shared homologies.: [5681] [1161] [736] [1881]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [9213]
Linear products..................: [9311] [9274] [9263] [9250] [8575] [8200] [7419] [6744] [6356] [3691] [2992] [2580] [98] [61] [50] [37]

In [24]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|1161bp_PCR_prod|38
|                           \/
|                           /\
|                           38|1881bp_PCR_prod|35
|                                              \/
|                                              /\
|                                              35|736bp_PCR_prod|61
|                                                                \/
|                                                                /\
|                                                                61-
|                                                                   |
 -------------------------------------------------------------------

In [29]:
pYPK0_PGI1tp_ScXKS1_FBA1tp = candidate.synced(pYPK0)

The pYPK0_PGI1tp_ScXKS1_FBA1tp has cseguid ```tDAwCS0RGOllmBII5oQTORuFXpw``` and 9213 bp.

In [30]:
print len(pYPK0_PGI1tp_ScXKS1_FBA1tp)
pYPK0_PGI1tp_ScXKS1_FBA1tp.cseguid()

9213


tDAwCS0RGOllmBII5oQTORuFXpw

In [31]:
pYPK0_PGI1tp_ScXKS1_FBA1tp.name="pMEC1129"
#pYPK0_PGI1tp_ScXKS1_FBA1tp.description = "pYPK0_PGI1tp_ScXKS1_FBA1tp"
pYPK0_PGI1tp_ScXKS1_FBA1tp.stamp()

cSEGUID_tDAwCS0RGOllmBII5oQTORuFXpw_2015-05-09T15:11:50.272874

In [36]:
pYPK0_PGI1tp_ScXKS1_FBA1tp.write("pYPK0_PGI1tp_ScXKS1_FBA1tp.gb")

##Download sequence

[pYPK0_PGI1tp_ScXKS1_FBA1tp](pYPK0_PGI1tp_ScXKS1_FBA1tp.gb)

In [1]:
reloaded = pydna.read("pYPK0_PGI1tp_ScXKS1_FBA1tp.gb")

In [2]:
reloaded.cseguid()

tDAwCS0RGOllmBII5oQTORuFXpw

In [3]:
reloaded.description

'pYPK0_PGI1tp_ScXKS1_FBA1tp cSEGUID_tDAwCS0RGOllmBII5oQTORuFXpw_2015-05-09T15:11:50.272874'